src/representor/base.ts

```diff
@@ -7,23 +7,96 @@ import Link from '../link';
  *
  * This is base class for a representation.
  */
-export default class Representation {
+export default abstract class Representation<T = string> {
 
-  body: any;
   contentType: string;
-  embedded: {
-    [uri: string]: object
-  };
-  links: Link[];
   uri: string;
 
-  constructor(uri: string, contentType: string, body: any) {
+  protected body: T;
+  protected links: LinkSet;
+
+  constructor(uri: string, contentType: string, body: string | null, headerLinks: LinkSet) {
 
     this.uri = uri;
     this.contentType = contentType;
+    this.links = headerLinks;
+    if (body !== null) {
+      this.setBody(this.parse(body));
+    }
+
+  }
+
+  getLink(rel: string): Link {
+
+    const links = this.links.get(rel);
+
+    if (!links) {
+      throw new LinkNotFound('Link with rel: ' + rel + ' not found on resource: ' + this.uri);
+    }
+
+    return links[0];
```

Comment:
zero-length links are truthy so this will return `undefined` instead of throw the LinkNotFound error.

Reply:
Missed this, thank you!